### ISSUE TITLE
isBoolean is not defined in firebase.auth()

### ISSUE THREAD

**Issue by Guarrakesh** [invertase/react-native-firebase#2321]
## Issue
Hi, I noticed that the implementation of auth().verifyPhoneNumber() has a "isBoolean" call which is not defined and causes impossibility to use the API.
Is this a bug with the code or a problem with language/library version?

---

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
platform :ios, '10.0'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'
require_relative '../node_modules/@react-native-firebase/app/pod_config'

target 'firebaselogin' do
  # Pods for firebaselogin

  # TODO: remove after RN >=0.60
  react_native_path = "../node_modules/react-native"
  pod 'React', :path => "#{react_native_path}", :subspecs => [
    # add more subspecs as needed, see: https://github.com/facebook/react-native/blob/0.59-stable/React.podspec
    'Core',
    'CxxBridge',
    'DevSupport',
    'RCTText',
    'RCTNetwork',
    'RCTImage',
    'RCTBlob',
    'RCTAnimation',
    'RCTWebSocket',
  ]

  pod "yoga", :path => "#{react_native_path}/ReactCommon/yoga"
  pod 'DoubleConversion', :podspec => "#{react_native_path}/third-party-podspecs/DoubleConversion.podspec"
  pod 'glog', :podspec => "#{react_native_path}/third-party-podspecs/glog.podspec"
  pod 'Folly', :podspec => "#{react_native_path}/third-party-podspecs/Folly.podspec"

  # TODO: add for RN >=0.60
  # pod 'React', :path => '../node_modules/react-native/'
  # pod 'React-Core', :path => '../node_modules/react-native/React'
  # pod 'React-DevSupport', :path => '../node_modules/react-native/React'
  # pod 'React-fishhook', :path => '../node_modules/react-native/Libraries/fishhook'
  # pod 'React-RCTActionSheet', :path => '../node_modules/react-native/Libraries/ActionSheetIOS'
  # pod 'React-RCTAnimation', :path => '../node_modules/react-native/Libraries/NativeAnimation'
  # pod 'React-RCTBlob', :path => '../node_modules/react-native/Libraries/Blob'
  # pod 'React-RCTImage', :path => '../node_modules/react-native/Libraries/Image'
  # pod 'React-RCTLinking', :path => '../node_modules/react-native/Libraries/LinkingIOS'
  # pod 'React-RCTNetwork', :path => '../node_modules/react-native/Libraries/Network'
  # pod 'React-RCTSettings', :path => '../node_modules/react-native/Libraries/Settings'
  # pod 'React-RCTText', :path => '../node_modules/react-native/Libraries/Text'
  # pod 'React-RCTVibration', :path => '../node_modules/react-native/Libraries/Vibration'
  # pod 'React-RCTWebSocket', :path => '../node_modules/react-native/Libraries/WebSocket'

  # pod 'React-cxxreact', :path => '../node_modules/react-native/ReactCommon/cxxreact'
  # pod 'React-jsi', :path => '../node_modules/react-native/ReactCommon/jsi'
  # pod 'React-jsiexecutor', :path => '../node_modules/react-native/ReactCommon/jsiexecutor'
  # pod 'React-jsinspector', :path => '../node_modules/react-native/ReactCommon/jsinspector'
  # pod 'yoga', :path => '../node_modules/react-native/ReactCommon/yoga'

  # pod 'DoubleConversion', :podspec => '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
  # pod 'glog', :podspec => '../node_modules/react-native/third-party-podspecs/glog.podspec'
  # pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'
pod 'RNFBAuth', :path => '../node_modules/@react-native-firebase/auth/ios'
  pod 'RNFBApp', :path => '../node_modules/@react-native-firebase/app'

  target 'firebaseloginTests' do
    inherit! :search_paths
    # Pods for testing
  end

  react_native_firebase!
  use_native_modules!
end

target 'firebaselogin-tvOS' do
  # Pods for firebaselogin-tvOS

  target 'firebaselogin-tvOSTests' do
    inherit! :search_paths
    # Pods for testing
  end

end

```

#### `AppDelegate.m`:

```objc
/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import "AppDelegate.h"

#import <React/RCTBridge.h>
#import <React/RCTBundleURLProvider.h>
#import <React/RCTRootView.h>
@import Firebase;

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  if ([FIRApp defaultApp] == nil) {
    [FIRApp configure];
  }
  RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self launchOptions:launchOptions];
  RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:bridge
                                                   moduleName:@"firebaselogin"
                                            initialProperties:nil];

  rootView.backgroundColor = [[UIColor alloc] initWithRed:1.0f green:1.0f blue:1.0f alpha:1];

  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  UIViewController *rootViewController = [UIViewController new];
  rootViewController.view = rootView;
  self.window.rootViewController = rootViewController;
  [self.window makeKeyAndVisible];
  return YES;
}

- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
{
#if DEBUG
  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
#else
  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
#endif
}

@end

```

</p>
</details>

---

### Android

<details><summary>Click To Expand</summary>
<p>

#### Have you converted to AndroidX?

- [ ] my application is an AndroidX application?
- [ ] I am using `android/gradle.settings` `jetifier=true` for Android compatibility?
- [ ] I am using the NPM package `jetifier` for react-native compatibility?

#### `android/build.gradle`:

```groovy
// N/A
```

#### `android/app/build.gradle`:

```groovy
// N/A
```

#### `android/settings.gradle`:

```groovy
// N/A
```

#### `MainApplication.java`:

```java
// N/A
```

#### `AndroidManifest.xml`:

```xml
<!-- N/A -->
```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
React Native Environment Info:
    System:
      OS: macOS 10.14.5
      CPU: (8) x64 Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
      Memory: 68.72 MB / 8.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 10.4.1 - /usr/local/bin/node
      Yarn: 1.7.0 - /usr/local/bin/yarn
      npm: 6.1.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.2, macOS 10.14, tvOS 12.2, watchOS 5.2
      Android SDK:
        API Levels: 23, 24, 25, 26, 27, 28
        Build Tools: 23.0.1, 24.0.1, 25.0.1, 26.0.1, 26.0.2, 26.0.3, 27.0.1, 27.0.3, 28.0.2, 28.0.3
        System Images: android-23 | Google APIs Intel x86 Atom_64, android-28 | Google APIs Intel x86 Atom
    IDEs:
      Android Studio: 3.1 AI-173.4907809
      Xcode: 10.2.1/10E1001 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.8.1 => 16.8.1 
      react-native: 0.59.8 => 0.59.8 
    npmGlobalPackages:
      react-native-cli: 2.0.1
      react-native-git-upgrade: 0.2.7
```

- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [ ] Android
  - [x ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - ^6.0.0-alpha.25
- **`Firebase` module(s) you're using that has the issue:**
  -  "@react-native-firebase/auth": "^6.0.0-alpha.8"
- **Are you using `TypeScript`?**
  - `Y`
  
</p>
</details>

---

Think `react-native-firebase` is great? Please consider supporting all of the project maintainers and contributors by donating via our [Open Collective](https://opencollective.com/react-native-firebase/donate) where all contributors can submit expenses. [[Learn More]](https://invertase.io/oss/react-native-firebase/contributing/donations-expenses)

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

**Comment by yogeverez** [comment]
There is a missing import of `isBoolean` validation function in auth/lib/index.js.
once you add it, you can call `verifyPhoneNumber()`.

https://github.com/invertase/react-native-firebase/blob/f429699990b6d521d996479f2bed96b52bbdb8d8/packages/auth/lib/index.js#L24

**Comment by Ehesp** [comment]
Just pushed this to master, thanks for flagging. Will get a release up soon.

We're going to work through test coverage of auth in the next couple of weeks, so these sort of things will get caught... sorry about that!

**Comment by Salakar** [comment]
This has now been published in 0.1.3 of the new packages, see https://github.com/invertase/react-native-firebase/issues/2025#issuecomment-519166641.

Thanks
